packages/fxa-settings/src/components/AppLayout/index.tsx

```diff
@@ -49,6 +50,18 @@ export const AppLayout = ({
 }: AppLayoutProps) => {
   const { l10n } = useLocalization();
   const config = useConfig();
+
+  // Sync the current page's layout preference to context so navigation
+  // preserves the layout during Suspense fallback, preventing visual flash.
+  // Only sync when not loading to avoid overwriting with fallback's layout.
+  // Use useLayoutEffect to update synchronously before browser paint.
+  const { setCurrentSplitLayout } = useLayoutContext();
+  useLayoutEffect(() => {
```

Comment:
Just curious, do you see flickering if this is `useEffect` instead of `useLayoutEffect`? I think it's fine for this use case, I just know React recommends not using it unless we _have_ to, so I figured I'd ask.

Reply:
Yes, I did swap from `useEffect` to `useLayoutEffect` here because of a flicker!